### PR TITLE
fix: override headers within req.rewrite()

### PR DESCRIPTION
**Pull request by taty2010**
## Description

Based on the description provided [here](https://github.com/netlify/pod-ecosystem-frameworks/issues/183) this update makes it to where we can now override the headers within the req.rewrite() like in the example below.

```js
    const res = await request.rewrite('/static-rewrite',
    {
      headers: {
        'x-rewrite-test': 'hello',
        'x-rewrite-test-2': 'hello-2'
      }
    })
```

## Tests

You can test this change yourself like so:

1. Go to https://deploy-preview-2189--next-plugin-edge-middleware.netlify.app/request-rewrite
2. review the headers within the Network tab and should see the headers from the example code above.

### CODE REVIEW

File: cypress/e2e/middleware/enhanced.cy.ts [hunk 1]
```diff
@@ -13,6 +13,12 @@ describe('Enhanced middleware', () => {
     })
   })
 
+  it.skip('passes in headers within request.rewrite()', () => {
```
Comment (MarcL) [inline cypress/e2e/middleware/enhanced.cy.ts:16]: Should this be skipped?

**Review by MarcL — COMMENTED** [review]
Think it just needs the test to be enabled and good to go.

File: cypress/e2e/middleware/enhanced.cy.ts (hunk 1 above)
Comment (taty2010) [reply cypress/e2e/middleware/enhanced.cy.ts]: ah this one should not be skipped, thanks!

**Review by nickytonline — APPROVED** [review]
Looks good! 🚢

**Pull request merged**